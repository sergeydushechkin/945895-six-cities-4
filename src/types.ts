enum CardType  {
  MAIN = `MAIN`,
  PROPERTY = `PROPERTY`,
  FAVORITES = `FAVORITES`,
}

enum Type {
  apartment = `apartment`,
  room = `room`,
  house = `house`,
  hotel = `hotel`,
}

enum SortTypes {
  POPULAR = `popular`,
  PRICE_LOW_HIGH = `priceLowHigh`,
  PRICE_HIGH_LOW = `priceHighLow`,
  TOP_RATED_FIRST = `topRatedFirst`
}

enum FavoritesImageSize {
  WIDTH = `150`,
  HEIGHT = `110`,
};

enum OthersImageSize {
  WIDTH = `260`,
  HEIGHT = `200`,
}

interface Offer {
  city: {
    name: string,
    coordinates: [number, number],
    zoom: number,
  }
  id: number,
  title: string,
  previewImage: string,
  price: number,
  type: Type,
  rating: number,
  isPremium: boolean,
  isFavorite: boolean,
  pictures: Array<string>,
  bedrooms: number,
  description: string,
  guests: number,
  features: Array<string>,
  host: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
  location: {
    coordinates: [number, number],
    zoom: number,
  },
}

interface Comment {
  comment: string,
  date: string,
  id: number,
  rating: number,
  user: {
    avatarUrl: string,
    id: number,
    isPro: boolean,
    name: string,
  },
}

interface AuthInfo {
  avatarUrl: string,
  email: string,
  id: number,
  isPro: boolean,
  name: string,
}

export {CardType, SortTypes, Type, Offer, Comment, AuthInfo, FavoritesImageSize, OthersImageSize};
