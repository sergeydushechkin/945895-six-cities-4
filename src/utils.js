const getRatingWidth = (rating) => {
  return Math.round(rating) * 20;
};

export {getRatingWidth};
