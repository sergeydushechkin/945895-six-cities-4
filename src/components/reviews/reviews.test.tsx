import * as React from "react";
import * as renderer from "react-test-renderer";

import {Comment} from "../../types";
import {noop} from "../../utils";
import Reviews from "./reviews";

const reviews: Array<Comment> = [
  {
    id: 1,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    rating: 4.3,
    date: `2019-04-24`,
    user: {
      avatarUrl: `img/1.png`,
      id: 4,
      isPro: false,
      name: `Max`,
    }
  }
];

it(`Render Reviews`, () => {
  const tree = renderer
    .create(
        <Reviews
          reviews={reviews}
          isUserLoggedIn={true}
          offerId={1}
          onPostComment={noop}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
