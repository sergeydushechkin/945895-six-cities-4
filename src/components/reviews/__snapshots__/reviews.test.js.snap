// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Reviews 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews Â· 
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="img/1.png"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Max
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "80%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.
        </p>
        <time
          className="reviews__time"
          dateTime="2019-04-24"
        >
          April 2019
        </time>
      </div>
    </li>
  </ul>
  <form
    action="#"
    className="reviews__form form"
    method="post"
    onSubmit={[Function]}
  >
    <label
      className="reviews__label form__label"
      htmlFor="review"
    >
      Your review
    </label>
    <div
      className="reviews__rating-form form__rating"
    >
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="5-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={5}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="5-stars"
        title="perfect"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="4-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={4}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="4-stars"
        title="good"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="3-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={3}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="3-stars"
        title="not bad"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="2-stars"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={2}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="2-stars"
        title="badly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
      <input
        checked={false}
        className="form__rating-input visually-hidden"
        disabled={false}
        id="1-star"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={1}
      />
      <label
        className="reviews__rating-label form__rating-label"
        htmlFor="1-star"
        title="terribly"
      >
        <svg
          className="form__star-image"
          height="33"
          width="37"
        >
          <use
            xlinkHref="#icon-star"
          />
        </svg>
      </label>
    </div>
    <textarea
      className="reviews__textarea form__textarea"
      disabled={false}
      id="review"
      name="review"
      onChange={[Function]}
      placeholder="Tell how was your stay, what you like and what can be improved"
      value=""
    />
    <div
      className="reviews__button-wrapper"
    >
      <p
        className="reviews__help"
      >
        To submit review please make sure to set 
        <span
          className="reviews__star"
        >
          rating
        </span>
         and describe your stay with at least 
        <b
          className="reviews__text-amount"
        >
          50 characters
        </b>
        .
      </p>
      <button
        className="reviews__submit form__submit button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
    
  </form>
</section>
`;
